jobs/worker.go

```diff
@@ -1,46 +1,84 @@
 package jobs
 
 import (
+	"context"
+	"errors"
 	"time"
 
 	l "github.com/RedHatInsights/sources-api-go/logger"
+	"github.com/RedHatInsights/sources-api-go/redis"
 )
 
-var ch chan Job
-
-// fire up a worker, initializing the channel to read from inline
-func Init() {
-	// initialize the channel to be used by the job worker. The `RunWorker`
-	// function just takes a channel so we can swap out the backing store for
-	// the jobs for anything we like as long as it can speak through a channel
-	// (maybe redis groups someday.)
-	ch = make(chan Job)
-	l.Log.Infof("Starting Worker for Delayed Jobs")
-	go RunWorker(ch)
+// the queue on redis we'll be sending the jobs to
+const workQueue = "sources_api_jobs"
+
+// Runs the worker just consuming off of a redis list
+func Run() {
+	l.Log.Infof("Starting up Background worker listening to redis queue %q", workQueue)
+
+	for {
+		// This is a BLocking Pop that will effectively wait forever until
+		// something gets queued.
+		//
+		// Once it does succeed - we check to see if the error is just
+		// `redis.Nil`, in which case things are fine and we continue.
+		val, err := redis.Client.BLPop(context.Background(), 0, workQueue).Result()
+		if err != nil {
+			if errors.Is(err, redis.Nil) {
+				l.Log.Debugf("nil value returned  from redis - continuing")
+			} else {
+				l.Log.Warnf("Failed to pop job from queue: %v", err)
```

Comment:
I would simply refactor this to `if !errors.Is(err, redis.Nil)`, since we don't care about them, do we?

Reply:
Ah yeah, great point. We don't really care if it was nil coming back or not.